lxd/response/response.go

```diff
@@ -327,9 +342,22 @@ func (r *errorResponse) String() string {
 	return r.msg
 }
 
-func (r *errorResponse) Render(w http.ResponseWriter, req *http.Request) error {
+func (r *errorResponse) Render(w http.ResponseWriter, req *http.Request) (err error) {
 	var output io.Writer
 
+	defer func() {
+		// Use the callback function to account the request for the metrics.
+		if err == nil && r.code >= 400 && r.code < 500 {
+			// 4* codes are considered client errors on HTTP.
+			request.MetricsCallback(req, metrics.ErrorClient)
+		} else {
+			// Any other status code here shoud be higher than or equal to 500 and is considered a server error.
+			// This also applies to the case where there was an error in writing the error response,
+			// independently of the code.
```

Comment:
> This also applies to the case where there was an error in writing the error response, independently of the code.

I'm not sure about this, whether we should consider those errors at all in the metrics, or whether they should be client errors. 

@mseralessandri how do you want errors in sending data from server to client to be considered in the metrics (if you want them considered at all), as there's no way to identity if its a client or server, or the network in between that is the issue.